### PR TITLE
Tag Cloud Block: Add outline style

### PR DESCRIPTION
**Pull request by scruffian**
## Description
This adds an outline style to the Tag Cloud block. We don't yet have an agreed standard for color names, so it's possible that these presets won't be defined in the theme.

## How has this been tested?
In TT2 and Skatepark

## Screenshots 
Screenshot 2021-12-03 at 10 42 51

Screenshot 2021-12-16 at 12 18 02
Screenshot 2021-12-16 at 12 17 44

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/tag-cloud/style.scss
Comment (mtias) [inline packages/block-library/src/tag-cloud/style.scss]: Does this need to be hard-coded?

Comment (scruffian) [reply packages/block-library/src/tag-cloud/style.scss]: We could make it a block attribute, but it wouldn't do anything unless this style was used.

File: packages/block-library/src/tag-cloud/style.scss
Comment (scruffian) [inline packages/block-library/src/tag-cloud/style.scss]: We might want to use some factor of block gap here.

File: packages/block-library/src/tag-cloud/style.scss
Comment (scruffian) [inline packages/block-library/src/tag-cloud/style.scss]: Should we remove the underline in this case?

File: packages/block-library/src/tag-cloud/style.scss
Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: It should only show up if the style is being used. We can also not set it to anything and let themes customize it for now.

Comment (scruffian) [reply packages/block-library/src/tag-cloud/style.scss]: Using a custom variable in theme.json?

File: packages/block-library/src/tag-cloud/style.scss
Comment (kjellr) [inline packages/block-library/src/tag-cloud/style.scss]: These variables don't exist by default, unfortunately. 

I'm not sure that we can specify a nice reversed-out hover state without tackling https://github.com/WordPress/twentytwentytwo/issues/99 first.

**Comment by kjellr** [comment]
I like this — it's a more modern version of a tag cloud. I think the only issue with it is figuring out how it could be implemented in a way that's lightweight and flexible enough to work out of the box for many themes. The current approach feels a bit too hardcoded to me.

File: packages/block-library/src/tag-cloud/style.scss
Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: I think we can avoid hover effects here

File: packages/block-library/src/tag-cloud/style.scss
Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: No, the theme can just target it with CSS directly if they want to

File: packages/block-library/src/tag-cloud/style.scss
Comment (kjellr) [reply packages/block-library/src/tag-cloud/style.scss]: Yeah, I don't think an underline is necessary.

File: packages/block-library/src/tag-cloud/style.scss
Comment (MaggieCabrera) [inline packages/block-library/src/tag-cloud/style.scss]: as much as I agree with this naming convention, we are assuming that foreground is going to be existing in the theme. I'd say at least give it a fallback when it's not the case?

Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: Can we go with just `currentColor` for border?

Comment (kjellr) [reply packages/block-library/src/tag-cloud/style.scss]: ^ Yeah, `currentColor` makes more sense here.

Comment (scruffian) [reply packages/block-library/src/tag-cloud/style.scss]: Done

File: packages/block-library/src/tag-cloud/style.scss
Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: Can it be a percentage so that if font-size is changed it works? Same for padding.

Comment (scruffian) [reply packages/block-library/src/tag-cloud/style.scss]: I don't think a percentage will work, but I updated it to use `ch` units which does. See the update screenshots above...

Comment (mtias) [reply packages/block-library/src/tag-cloud/style.scss]: Sweet!

File: packages/block-library/src/tag-cloud/style.scss
Comment (kjellr) [inline packages/block-library/src/tag-cloud/style.scss]: The [normal font size has been depreciated](https://github.com/WordPress/gutenberg/pull/37381), so we might want to switch to "Medium" here? 

.... but if we don't specify a font size maybe we're better off? Then it should inherit the default size.

Comment (scruffian) [reply packages/block-library/src/tag-cloud/style.scss]: The problem is if we don't specify a size then the font sizes are determined by how often the tag is used

**Comment by MaggieCabrera** [comment]
I'm trying this on emptytheme and the frontend doesn't reflect what's in the editor:

Editor:

Screenshot 2021-12-16 at 14 26 02

Frontend:

Screenshot 2021-12-16 at 14 26 13

File: packages/block-library/src/tag-cloud/style.scss
Comment (MaggieCabrera) [reply packages/block-library/src/tag-cloud/style.scss]: `unset` will do that for you

**Comment by scruffian** [comment]
It works for me in emptytheme:
Screenshot 2021-12-16 at 20 22 29

**Comment by kjellr** [comment]
It's looking right in the front end for me. 👍 

In general, this seems solid to me. The code is generic enough that it inherits all it can from the theme its placed in. Nice work! 

The only remaining comment on my end is that I think we should reconsider the "pill" name since these aren't rounded anymore. Maybe we could use "Button" or "Outline" instead? Outline might be more reasonable, since folks would assume a "Button" would match the button block.

**Comment by scruffian** [comment]
> The only remaining comment on my end is that I think we should reconsider the "pill" name since these aren't rounded anymore. Maybe we could use "Button" or "Outline" instead? Outline might be more reasonable, since folks would assume a "Button" would match the button block.

Done!

**Review by kjellr — APPROVED** [review]
This looks good, thanks @scruffian!

**Comment by scruffian** [comment]
It would be good to have another review before we merge so this doesn't end up needing a revert.

cc @youknowriad @mcsf

**Comment by youknowriad** [comment]
I have no objections here as long as designers are happy

File: packages/block-library/src/tag-cloud/index.php
Comment (gziolo) [inline packages/block-library/src/tag-cloud/index.php:55]: Any particular reason why this style isn't included in the `block.json` file? See: https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#block-styles.

File: packages/block-library/src/tag-cloud/index.php
Comment (gziolo) [inline packages/block-library/src/tag-cloud/index.php:60]: I don't see a style registered with the handle `outline` in this PR and the styles were added to the existing CSS file, so maybe we don't need that part.

**Pull request merged**
